### ISSUE TITLE
units are not updated after a conversion

### ISSUE THREAD

**Issue by haussel** [astropy/astropy#3456]
Hello, 

I needed to converted the critical density in cosmology to Msun/Mpc^3 and stumbled on a problem:

I run python 2.7.9 and IPython 2.3.1

```python
import numpy as np
print "numpy version " + np.__version__
import astropy
import astropy.units as u
print "astropy version " + astropy.__version__
from astropy.cosmology import Planck13

msun_per_mpc3 = u.solMass / u.Mpc / u.Mpc / u.Mpc
g_per_cm3 = u.g / u.cm / u.cm / u.cm
gcm3_to_mpc3 =  g_per_cm3.get_converter(msun_per_mpc3)
print Planck13.critical_density(1)
print gcm3_to_mpc3(cosmo.critical_density(1))
```

The output is: 
numpy version 1.9.1
astropy version 0.4.4
2.7270262927e-29 g / cm3
4.01453517314e+11 g / cm3

While the values are correct, the units are still shown as g/cm3...

**Comment by mhvk** [comment]
@haussel - I think the way I would recommend to do conversions like yours is
```
In [4]: Planck13.critical_density(1).to(u.solMass / u.Mpc / u.Mpc / u.Mpc)
Out[4]: <Quantity 402795252815.4388 solMass / Mpc3>
```
(Of course, if you use this quantity later in a calculation, its units do not matter! it will be converted automatically.)

The `get_converter` method is meant to be used for plain values or arrays, not quantities. That it does not raise an exception, however, would seem to be a bug. cc @mdboom

**Comment by haussel** [comment]
Ok, re-reading the doc, I think I see your point. But I would make it more clear in the documentation, and maybe have the functions that are given by the get_converter() method only output values.

**Comment by astrofrog** [comment]
@mhvk - I agree that the current behavior is a bug and is because ``get_converter`` pre-dates the whole ``Quantity`` framework. One could imagine making it able to handle quantities without raising an exception though, right?

**Comment by mhvk** [comment]
@astrofrog - yes, probably it is better to just make `get_converter` deal with quantities properly

@haussel - is there anything specific in the documentation that we could improve? (It is hard to see what is unclear once you have used the code too much!)

**Comment by haussel** [comment]
@mhvk : the example in http://docs.astropy.org/en/stable/units/conversion.html#obtaining-a-conversion-function does apply the `get_converter` fonction to an array, not a quantity. I think it would be clearer if the sentence:
> Conversion to different units involves obtaining a conversion function and then applying it to the value, or values to be converted.

would read instead:

> Conversion to different units involves obtaining a conversion function and then applying it to the value, or values to be converted*, not to quantities whose units will not be updated.*

**Comment by mdboom** [comment]
Personally, I think we should just deprecate and privatize `get_converter` -- making it handle Quanitities will just make everything slower in an important inner loop.  We already document it on a page of things that aren't for general use...

**Comment by mhvk** [comment]
@mdboom - I agree.  Which probably is a reason to get in a quick PR...  Especially as I've just been arguing in #3459 that units is now stable & mature.

**Comment by mdboom** [comment]
Closed by #3474 -- by deprecating `get_converter`.  Use `Unit.to` or `Quantity.to` instead.
